### DISCUSSION TITLE
Use GLib macros for some __attribute__ usage?

### DISCUSSION THREAD

**Discussion by tintou** [libvips/libvips#2869] [Ideas]
Instead of redefining the `__attribute__` macro, it would be nice to use the available macros in GLib.
In particular:
 * [G_GNUC_NULL_TERMINATED](https://developer-old.gnome.org/glib/2.68/glib-Miscellaneous-Macros.html#G-GNUC-NULL-TERMINATED:CAPS)
 * [G_GNUC_PRINTF](https://developer-old.gnome.org/glib/2.68/glib-Miscellaneous-Macros.html#G-GNUC-PRINTF:CAPS)

**Comment by jcupitt** [comment]
Hi @tintou, good idea, I made an enhancement issue.

**Reply by jcupitt** [reply]
https://github.com/libvips/libvips/issues/2871
